content/blog/2022/2022-02-14-spring-shell.md

```diff
@@ -0,0 +1,295 @@
+
+We can specify the name of the parameter using the `value` property of the `@ShellOption` annotation.
+
+If we don't specify the value however, Spring Shell assigns it a default value of parameter name "-" separated prefixed by `ShellMethod.prefix()`.
+- The `ShellMethod.prefix()` default value is "--" unless changed in the `@ShellMethod` annotation:
+```java
+@ShellMethod(key = "my-ssh", prefix = "-", value = "connect to remote server")  
+public void ssh(@ShellOption String remoteServer)  
+{  
+    log.info(format("Logged to machine '%s'", remoteServer));  
+}
+```
+- In the above example the parameter name will be `-remote-server`.
+
+#### Default Parameters Values
+We can assign default values to parameters in case the user doesn't specify any. Doing this also allows the user to treat those parameters as optional:
+```java
+@ShellMethod(value = "connect to remote server")  
+public void ssh(@ShellOption(value = "--s", defaultValue = "default-server") String remoteServer)  
+{  
+    log.info(format("Logged to machine '%s'", remoteServer));  
+}
+```
+Typing only `ssh` to the console will give us:
+```bash
+shell:>ssh
+2022-02-11 19:55:05.133  INFO 4700 --- [           main] j.t.springshell.command.SSHCommand       : Logged to machine 'default-server'
+```
+
+#### Multi-valued Parameters
+We can specify multiple values for a single parameter by using the `arity()` attribute of the `@ShellOption` annotation. Simply use a collection or array for the parameter type, and specify how many values are expected":
+```java
+@ShellMethod(value = "add keys")  
+public void sshAdd(@ShellOption(value = "--k", arity = 2) String[] keys)  
+{  
+    log.info(format("Adding keys '%s' '%s'", keys[0], keys[1]));  
+}
+```
+
+Let's try the command out in the shell:
+
+```bash
+shell:>ssh-add --k test1 test2
+2022-02-12 18:27:00.301  INFO 4928 --- [           main] j.t.springshell.command.SSHCommand       : Adding keys 'test1' 'test2'
+```
+
+#### Boolean Parameters
+Boolean parameters receive a special treatment by command-line utilities. The absence of the parameter in the command indicates a false value. On the other hand, its existence indicates true value:
+```java
+@ShellMethod(value = "sign in")  
+public void sshLogin(@ShellOption(value = "--r") boolean rememberMe)  
+{  
+    log.info(format("remember me option is '%s'", rememberMe));  
+}
+```
+Let's check it out in the command line:
+```bash
+shell:>ssh-login --r
+2022-02-12 18:41:34.903  INFO 10044 --- [           main] j.t.springshell.command.SSHCommand       : remember me option is 'true'
+shell:>ssh-login
+2022-02-12 18:41:44.606  INFO 10044 --- [           main] j.t.springshell.command.SSHCommand       : remember me option is 'false'
+```
+#### Validating Command Parameters
+According to the Spring Shell team
+
+> Spring Shell integrates with the  [Bean Validation API](https://beanvalidation.org/)  to support automatic and self documenting constraints on command parameters.
```

Comment:
Again, please rephrase in your own words. You should link out to the docs in the conclusion or intro, however.

Reply:
Maybe also mention that you have to add the bean validation starter as a dependency.